geopandas/geodataframe.py

```diff
@@ -1419,6 +1419,16 @@ def __getitem__(self, key):
         return a GeoDataFrame.
         """
         result = super().__getitem__(key)
+        # Custom logic to avoid waiting for pandas GH51895
+        # result is not geometry dtype for multi-indexes
+        if (
+            isinstance(result, Series)
+            and isinstance(self.columns, pd.MultiIndex)
+            and not is_geometry_type(result)
+        ):
+            loc = self.columns.get_loc(key)
+            # squeeze stops multilevel columns from returning a gdf
+            result = self.iloc[:, loc].squeeze()
```

Comment:
`squeeze` might do the wrong thing in the corner case of a single row dataframe ..?